x/concentrated-liquidity/lp_test.go

```diff
@@ -69,43 +79,65 @@ func (s *KeeperTestSuite) TestCreatePosition() {
 			tc := tc
 			s.SetupTest()
 
-			// Merge tc with baseCase and update tc
-			// to the merged result. This is done
-			// to reduce the amount of boilerplate
-			// in test cases.
+			// Merge tc with baseCase and update tc to the merged result. This is done to reduce the amount of boilerplate in test cases.
 			baseConfigCopy := *baseCase
 			mergeConfigs(&baseConfigCopy, &tc)
 			tc = baseConfigCopy
 
-			_, err := s.App.ConcentratedLiquidityKeeper.CreateNewConcentratedLiquidityPool(s.Ctx, tc.poolId, denom0, denom1, tc.currentSqrtP, tc.currentTick)
+			// Create a CL pool with poolId 1
+			pool, err := s.App.ConcentratedLiquidityKeeper.CreateNewConcentratedLiquidityPool(s.Ctx, 1, ETH, USDC, tc.currentSqrtP, tc.currentTick)
 			s.Require().NoError(err)
 
+			// Fund test account and create the desired position
 			s.FundAcc(s.TestAccs[0], sdk.NewCoins(sdk.NewCoin("eth", sdk.NewInt(10000000000000)), sdk.NewCoin("usdc", sdk.NewInt(1000000000000))))
+
+			// Note user and pool account balances before create position is called
+			userBalancePrePositionCreation := s.App.BankKeeper.GetAllBalances(s.Ctx, s.TestAccs[0])
+			poolBalancePrePositionCreation := s.App.BankKeeper.GetAllBalances(s.Ctx, pool.GetAddress())
+
 			asset0, asset1, liquidityCreated, err := s.App.ConcentratedLiquidityKeeper.CreatePosition(s.Ctx, tc.poolId, s.TestAccs[0], tc.amount0Desired, tc.amount1Desired, tc.amount0Minimum, tc.amount1Minimum, tc.lowerTick, tc.upperTick)
 
+			// Note user and pool account balances to compare after create position is called
+			userBalancePostPositionCreation := s.App.BankKeeper.GetAllBalances(s.Ctx, s.TestAccs[0])
+			poolBalancePostPositionCreation := s.App.BankKeeper.GetAllBalances(s.Ctx, pool.GetAddress())
+
+			// If we expect an error, make sure:
+			// - some error was emitted
+			// - asset0 and asset1 that were calculated from create position is nil
+			// - the error emitted was the expected error
+			// - account balances are untouched
 			if tc.expectedError != nil {
 				s.Require().Error(err)
 				s.Require().Equal(asset0, sdk.Int{})
 				s.Require().Equal(asset1, sdk.Int{})
-				s.Require().ErrorAs(err, &tc.expectedError)
+				s.Require().ErrorContains(err, tc.expectedError.Error())
```

Comment:
q: Why did `ErrorAs` not work anymore?

Reply:
Oh it does, just changed it back (was having issues not realizing it was supposed to be a pointer)